group-coordinator/src/test/java/org/apache/kafka/coordinator/group/modern/SubscribedTopicMetadataTest.java

```diff
@@ -79,4 +80,30 @@ public void testEquals() {
         topicMetadataMap2.put(topicId, new TopicMetadata(topicId, "newTopic", 5));
         assertNotEquals(new SubscribedTopicDescriberImpl(topicMetadataMap2), subscribedTopicMetadata);
     }
+
+    @Test
+    public void testAssignablePartitions() {
+        // null allow map (all partitions assignable)
+        subscribedTopicMetadata = new SubscribedTopicDescriberImpl(topicMetadataMap, null);
+        String t1Name = "t1";
+        Uuid t1Id = Uuid.randomUuid();
+        topicMetadataMap.put(t1Id, new TopicMetadata(t1Id, t1Name, 5));
+        assertEquals(Set.of(0, 1, 2, 3, 4), subscribedTopicMetadata.assignablePartitions(t1Id));
+
+        // empty allow map (nothing assignable)
+        subscribedTopicMetadata = new SubscribedTopicDescriberImpl(topicMetadataMap, Map.of());
+        assertEquals(Set.of(), subscribedTopicMetadata.assignablePartitions(t1Id));
+
+        // few assignable partitions
+        subscribedTopicMetadata = new SubscribedTopicDescriberImpl(topicMetadataMap, Map.of(
```

Comment:
nit: Slightly odd indentation with the line break in the middle of the deepest method call. Similar on line 104.